core/capabilities/ccip/oraclecreator/bootstrap.go

```diff
@@ -232,13 +239,16 @@ type peerGroupDialer struct {
 	oraclePeerIDs      []ragep2ptypes.PeerID
 	commitConfigDigest [32]byte
 
+	// state: accessed in mutually exclusive fashion
+	// between Close() and syncLoop().
+	// Close shuts down syncLoop() first and then cleans up these.
 	activePeerGroups            []networking.PeerGroup
 	activeEndpointConfigDigests []cciptypes.Bytes32
 
 	syncInterval time.Duration
 
-	mu         *sync.Mutex
```

Comment:
Mutex is no longer needed since mutually exclusive access to `activePeerGroups` and `activeEndpointConfigDigests` is achieved by the `stopChan` and `wg`